packages/contract/src/DaimoAccount.sol

```diff
@@ -161,14 +181,36 @@ contract DaimoAccount is IAccount, UUPSUpgradeable, Initializable, IERC1271 {
     ) private view returns (bool) {
         if (signature.length < 1) return false;
 
-        // First bit identifies the keySlot
-        uint8 keySlot = uint8(signature[0]);
+        // First bit identifies the number of signatures
+        uint8 numSignatures = uint8(signature[0]);
+        if (numSignatures < signatureThreshold) return false;
+
+        // TODO: this requires all signatures be valid. What
+        // if 2 sigs are required, 3 get passed in, and 1 is
+        // invalid. Should op succeed?
+        uint256 offset = 1;
+        for (uint256 i = 0; i < numSignatures;) {
+            uint8 keySlot = uint8(signature[offset]);
```

Comment:
[reposting from gh so it doesn't get lost] need to add check to make sure keys are re-used

pseudo code that ~works gas-efficiently since we don't have access to maps in memory:
```
        uint256 usedKeys;
        loop i = each used keyslot {
          uint256 bit = 1 << i;
          uint256 flipped = usedKeys ^= bit;
          if (flipped & bit == 0) revert Invalid();
        }
```